qiskit_algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -67,15 +69,18 @@ def __init__(
                 ``num_timesteps`` and an evaluation of :attr:`.TimeEvolutionProblem.aux_operators`
                 at every time-step. If ``reps`` is larger than 1, the true number of time-steps will
                 be ``num_timesteps * reps``.
-            num_timesteps: The number of time-steps the full evolution time is divided into
-                (repetitions of ``product_formula``)
             estimator: An estimator primitive used for calculating expectation values of
                 ``TimeEvolutionProblem.aux_operators``.
+            num_timesteps: The number of time-steps the full evolution time is divided into
+                (repetitions of ``product_formula``).
+            insert_barriers: If True, insert a barrier after the initial state and after each Trotter
+                step.
         """
 
         self.product_formula = product_formula
         self.num_timesteps = num_timesteps
         self.estimator = estimator
+        self.insert_barriers = insert_barriers
```

Comment:
```suggestion
        self._insert_barriers = insert_barriers
```

I am thinking maybe this should be a private instance variable that the parameter value is assigned to. `product_formula`, `num_timesteps` and `estimator` above are all public setter/getters and the assignment is done via the documented setter to some private instance var.

As it stands self.insert_barriers is adding an undocumented variable, that you can later directly set/get, to the public API. I am not sure we really need the ability to update this after the construction and hence am thinking this should be simply changed to have it marked private as per the suggestion. If its felt strongly that it should be updateable then it ought to have a setter/getter like the others - but I don't personally think this functionality warrants that.